app/src/main/java/com/example/android/sunshine/app/ForecastAdapter.java

```diff
@@ -2,57 +2,86 @@
 
 import android.content.Context;
 import android.database.Cursor;
-import android.support.v4.widget.CursorAdapter;
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.ImageView;
 import android.widget.TextView;
 
+import com.bignerdranch.android.multiselector.MultiSelector;
+import com.bignerdranch.android.multiselector.SingleSelector;
+import com.bignerdranch.android.multiselector.SwappingHolder;
+import com.bumptech.glide.Glide;
 import com.example.android.sunshine.app.data.WeatherContract;
 
-import com.bumptech.glide.Glide;
+import java.util.List;
 
 /**
  * {@link ForecastAdapter} exposes a list of weather forecasts
- * from a {@link Cursor} to a {@link android.widget.ListView}.
+ * from a {@link Cursor} to a {@link RecyclerView}.
  */
-public class ForecastAdapter extends CursorAdapter {
+public class ForecastAdapter extends RecyclerView.Adapter<ForecastAdapter.ViewHolder> {
 
-    private static final int VIEW_TYPE_COUNT = 2;
     private static final int VIEW_TYPE_TODAY = 0;
     private static final int VIEW_TYPE_FUTURE_DAY = 1;
+    private final Context context;
+    private final OnClickHandler onClickHandler;
+    private final MultiSelector selector;
 
     // Flag to determine if we want to use a separate view for "today".
-    private boolean mUseTodayLayout = true;
-
-    /**
-     * Cache of the children views for a forecast list item.
-     */
-    public static class ViewHolder {
-        public final ImageView iconView;
-        public final TextView dateView;
-        public final TextView descriptionView;
-        public final TextView highTempView;
-        public final TextView lowTempView;
-
-        public ViewHolder(View view) {
+    private boolean useTodayLayout = true;
+    private Cursor cursor;
+
+    static class ViewHolder extends SwappingHolder implements View.OnClickListener {
+
+        final ImageView iconView;
+        final TextView dateView;
+        final TextView descriptionView;
+        final TextView highTempView;
+        final TextView lowTempView;
```

Comment:
Yes, they are used in the outer class, and I want to avoid the creation of synthetic accessor methods. See https://twitter.com/JakeWharton/status/679018449406992384 as a reference :D